src/poetry/utils/env/python_manager.py

```diff
@@ -0,0 +1,163 @@
+    from poetry.poetry import Poetry
+
+
+class Python:
+    def __init__(self, executable: str | Path, version: Version | None = None) -> None:
+        self.executable = Path(executable)
+        self._version = version
+
+    @property
+    def version(self) -> Version:
+        if not self._version:
+            if self.executable == Path(sys.executable):
+                _python_version = ".".join(str(v) for v in sys.version_info[:3])
+            else:
+                encoding = "locale" if sys.version_info >= (3, 10) else None
+                _python_version = decode(
+                    subprocess.check_output(
+                        [str(self.executable), "-c", GET_PYTHON_VERSION_ONELINER],
+                        text=True,
+                        encoding=encoding,
+                    ).strip()
+                )
+            self._version = Version.parse(_python_version)
+
+        return self._version
+
+    @cached_property
+    def patch_version(self) -> Version:
+        return Version.from_parts(
+            major=self.version.major,
+            minor=self.version.minor,
+            patch=self.version.patch,
+        )
+
+    @cached_property
+    def minor_version(self) -> Version:
+        return Version.from_parts(major=self.version.major, minor=self.version.minor)
+
+    @staticmethod
+    def _full_python_path(python: str) -> Path | None:
+        # eg first find pythonXY.bat on windows.
+        path_python = shutil.which(python)
+        if path_python is None:
+            return None
+
+        try:
+            encoding = "locale" if sys.version_info >= (3, 10) else None
+            executable = subprocess.check_output(
+                [path_python, "-c", "import sys; print(sys.executable)"],
+                text=True,
+                encoding=encoding,
+            ).strip()
+            return Path(executable)
+
+        except subprocess.CalledProcessError:
+            return None
+
+    @staticmethod
+    def _detect_active_python(io: IO) -> Path | None:
+        io.write_error_line(
+            "Trying to detect current active python executable as specified in"
+            " the config.",
+            verbosity=Verbosity.VERBOSE,
+        )
+
+        executable = Python._full_python_path("python")
+
+        if executable is not None:
+            io.write_error_line(f"Found: {executable}", verbosity=Verbosity.VERBOSE)
+        else:
+            io.write_error_line(
+                "Unable to detect the current active python executable. Falling"
+                " back to default.",
+                verbosity=Verbosity.VERBOSE,
+            )
+
+        return executable
+
+    @classmethod
+    def get_system_python(cls) -> Python:
+        return cls(executable=sys.executable)
+
+    @classmethod
+    def get_by_name(cls, python_name: str) -> Python | None:
+        _executable = cls._full_python_path(python_name)
+        if not _executable:
+            return None
```

Comment:
```suggestion
        if not (_executable := cls._full_python_path(python_name)):
            return None
```

also, not sure if we need the `_` prefix there

Reply:
At work, we consider it bad style to use the walrus operator in an `if` statement even though the assigned variable is used outside of the `if` later. In other words, I do not like the suggested change.